### DISCUSSION TITLE
Struggling with include path on macos

### DISCUSSION THREAD

**Discussion by bentxt** [thradams/cake#198] [General]
Hi,
After running a simple transformation example to make sure `cake` runs, I'd like to take it a little bit further and run an example with includes, like this one here:

```
#include <stdlib.h>
#include <stdio.h>

int main() {
    int n = 2;
    int m = 3;
    int (*p)[n][m] = malloc(sizeof * p);

    printf("%zu\n", sizeof(*p));

    free(p);
}
```

### with include from Apple SDK
when I use this path in 'cakeconfig.h':
```
#pragma dir "/Library/Developer//CommandLineTools/SDKs/MacOSX14.4.sdk/usr/include"
```

I get this error:
```
./cake  -target=c11 const.c
Cake 0.9.17
../cake/src/const.c
/Library/Developer/CommandLineTools/SDKs/MacOSX14.4.sdk/usr/include/sys/_types.h:55:25: error: expected TK_SEMICOLON [E970]
 55 |typedef __int64_t       __darwin_blkcnt_t;      /* total blocks */
    |                        ~~~~~~~~~~~~~~~~~
 1 files in 0.06 seconds
 1 errors 8 warnings 0 notes
```

### with include from clang

when I use this path in 'cakeconfig.h':
```
#pragma dir "/usr/local/Cellar/llvm/18.1.8/include/c++/v1"
```

I get this error:
```
./cake  -target=c11 const.c
Cake 0.9.17
[..]/const.c
/Volumes/scand500gb01/storeboxes_ben_drive01_scand500gb01/lan-store_ben_manual_drive01/Builds/cake/cake/src/const.c:7:22: error: not found 'malloc' [E680]
 7 |    int (*p)[n][m] = malloc(sizeof * p);
   |                     ~~~~~~
[..]/const.c:7:22: error: expected TK_SEMICOLON [E970]
 7 |    int (*p)[n][m] = malloc(sizeof * p);
   |                     ~~~~~~

 1 files in 0.01 seconds
 2 errors 0 warnings 0 notes
```

I admit and I'm a little bit sorry for not having a full understanding of what going on here, but still maybe somebody can help.

thanks
Ben

**Comment by thradams** [comment]
Type `__int64_t` is not recognized by cake. 
In such cases I generally clone  the compiler, in this case I can add __int64_t. (I guess clang has it on mac)

The workaround for now it define it on cake config.

typedef long long __int64_t;

In general MacOS is not tested.

OBS:
VLAs
int (*p)[n][m] = malloc(sizeof * p);

Are not implemented yet.
http://thradams.com/cake/manual.html#toc_14

**Reply by bentxt** [reply]
thanks alot for your help?

Do you think it would be possible to ignore includes all together?

**Comment by thradams** [comment]
I was planning to create independent  header files for cake. 

I started here
https://github.com/thradams/cake/tree/main/src/include
but I didn't finished.

The advantage of independent headers is that cake would always be working, specially 
for standard code. It could help someone that just want to try cake.

The disadvantage is that some not standard headers could be missing.

If you are planning to do basic tests you can create the headers with the declarations.

The online version 
http://thradams.com/cake/playground.html
has it own headers and can help you to write your headers.

For instance

http://thradams.com/cake/playground.html?code=I2luY2x1ZGUgPHN0ZGlvLmg%2B&to=-1&options=-E

The problem of this method is that is removes macros.


The headers files used by the web version are inside the code at

https://github.com/thradams/cake/blob/main/src/fs.c

So ideally I should have the cake "includes" and use the same files for the web version.
(C23 #embed could be useful for that )
